### PR TITLE
turn on async snowflake

### PR DESCRIPTION
**Pull request by cgardens**
## What
* This is the feature branch for Async Snowflake. It "turns on" the Async Snowflake Destination. 
* We want to incrementally roll this destination to users. In order to do that we want all the code for Async Snowflake can live in master and that all we need is this single commit to "turn it on". In other words, this PR should only ever differ from master by one commit. Then we will use this to publish the Async Snowflake Destination during roll out. Once it is 100% rolled out, this PR can be merged into master. 
* Basically this command is how we update it: `git rebase --onto origin/master HEAD~1`

## How
*Describe the solution*

## Recommended reading order
1. `SnowflakeDestination.java` <== this is the only file that should ever have a diff. if it doesn't, there's a problem!

## 🚨 User Impact 🚨
Yes, this officially switched destination-snowflake over to the async version.

### CODE REVIEW

**Comment by ryankfu** [comment]
/publish connector=connectors/destination-snowflake pre-release=true

**Comment by ryankfu** [comment]
/legacy-publish connector=connectors/destination-snowflake pre-release=true run-tests=false

**Comment by davinchia** [comment]
No longer needed.
